### ISSUE TITLE
About Video FPS

### ISSUE THREAD

**Issue by ferro07** [ultralytics/yolov5#1799]
Hey Dear @glenn-jocher 
When I print FPS in Detect.py for mp4 video, it is always 30 fps for all models (s, m, l, x) , even with different inference img-size.
Only inference time is different (different img-size). 
Is there somehing I miss, or I am on wron way?
Many Thanks...

**Comment by wudashuo** [comment]
Because your original video is 30 FPS?
Read the code:
https://github.com/ultralytics/yolov5/blob/73cf75faa848cb6c7cd8ff1b192ad341098c042c/detect.py#L134
And I think maybe you misunderstood about FPS, if you want to know the performance about the model, you should not find the FPS from the inference video, it's different. Video consists of many image frames, and the inference video's  FPS should be consistent with the original video.

**Comment by jaqub-manuel** [comment]
Dear @wudashuo, thanks for quick reply. You are right, my viode is 30 fps. Now, I have tested yolov4 and yolov5 for same vidoes. 
first video 30fps 212 frame hd 1920x1080 (30fps), yolov4.cf (512x512) = 71.4 fps,
first video 30fps 212 frame hd 1920x1080 (30fps), yolov5m (512x512) = 0.010s per image, and total time 2.771s, avg inference = 2771s/212 frame 0.013, and 1/T (as @glenn-jocher indicated in yolov3 repo)= ~77 fps ?, but printing fps in detect.py = 29.82 (may be some delay via buffer etc.),

second video 60fps 212 frame hd 1920x1080 (30fps), yolov4.cf (512x512) = 70.5 fps,
second video 60fps 212 frame hd 1920x1080 (30fps), yolov5m (512x512) = 5.540s total inference, avg inf= 5540/212frame= 0.026, fps= 38fps, (in detect.py, printing fps=60)

I have calculated like above...How to calculate FPS from video in yolov5,...YoloV4 (darknet), calcualtes fine, but I am not sure about yolov5..
Many thanks.

**Comment by jaqub-manuel** [comment]
Dear @wudashuo, thanks for quick reply. You are right, my viode is 30 fps. Now, I have tested yolov4 and yolov5 for same videos.
**first video 30fps 212 frame hd 1920x1080 (30fps), yolov4.cfg (512x512)** = 71.4 fps, (last fps)
**first video 30fps 212 frame hd 1920x1080 (30fps), yolov5m (512x512)** = las inference= 0.009s per image, and total time 2.771s, avg inference = 2771s/212 frame 0.013, and (FPs from inference = 1/T as @glenn-jocher indicated in yolov3 repo)= ~77 fps ?, last fps = 111FPs but printing fps in detect.py = 29.82 (may be some delay via buffer etc.),

**second video 60fps 212 frame hd 1920x1080 (60fps), yolov4.cfg (512x512)** = 70.5 fps, (last fps)
**second video 60fps 212 frame hd 1920x1080 (60fps), yolov5m (512x512)** = 5.540s total inference, avg inf= 5540/212frame= 0.026, fps= 38fps, last inference = 0.010s, = last fps =100FPs (in detect.py, printing fps=60)

I have calculated like above...How to calculate FPS from video in yolov5,...YoloV4 (darknet), calcualtes FPS and avg FPS, but I am not sure about yolov5..
Many thanks.

**Comment by ferro07** [comment]
Dear @wudashuo, thanks for quick reply. You are right, my viode is 30 fps. Now, I have tested yolov4 and yolov5 for same videos.
**first video 30fps 212 frame hd 1920x1080 (30fps), yolov4.cfg (512x512)** = 71.4 fps, (last fps)
**first video 30fps 212 frame hd 1920x1080 (30fps), yolov5m (512x512)** = las inference= 0.009s per image, and total time 2.771s, avg inference = 2771s/212 frame 0.013, and (FPs from inference = 1/T as @glenn-jocher indicated in yolov3 repo)= ~77 fps ?, last fps = 111FPs but printing fps in detect.py = 29.82 (may be some delay via buffer etc.),

**second video 60fps 212 frame hd 1920x1080 (60fps), yolov4.cfg (512x512)** = 70.5 fps, (last fps)
**second video 60fps 212 frame hd 1920x1080 (60fps), yolov5m (512x512)** = 5.540s total inference, avg inf= 5540/212frame= 0.026, fps= 38fps, last inference = 0.010s, = last fps =100FPs (in detect.py, printing fps=60)

I have calculated like above...How to calculate FPS from video in yolov5,...YoloV4 (darknet), calcualtes FPS and avg FPS, but I am not sure about yolov5..
Many thanks.
